### PR TITLE
Home Assistant custom component

### PR DESCRIPTION
**Pull request by robbrad**

### CODE REVIEW

File: custom_components/uk_bin_collection/__init__.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up UK Bin Collection Data from a config entry."""
+    _LOGGER.info(LOG_PREFIX + "Data Supplied: %s", entry.data)
+    council_name = entry.data.get("council", "unknown council")
+    _LOGGER.info(
+        LOG_PREFIX + "Setting up UK Bin Collection Data for council: %s", council_name
+    )
+
+    # coordinator = UKBinCollectionDataUpdateCoordinator(hass, entry.data)
```
Comment (OliverCullimore) [inline custom_components/uk_bin_collection/__init__.py]: Could probably remove this commented line?

File: uk_bin_collection/uk_bin_collection/councils/CheshireEastCouncil.py
Comment (OliverCullimore) [inline uk_bin_collection/uk_bin_collection/councils/CheshireEastCouncil.py:1]: Did we mean to be updating a council as part of merging this?

File: custom_components/uk_bin_collection/__init__.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    hass.data.setdefault(DOMAIN, {})
+    # hass.data[DOMAIN][entry.entry_id] = coordinator
```
Comment (OliverCullimore) [inline custom_components/uk_bin_collection/__init__.py]: Could probably remove this commented line too?

File: custom_components/uk_bin_collection/config_flow.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+
+class UkBinCollectionConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    def __init__(self):
+        self.councils_data = None
+
+    async def get_councils_json(self) -> object:
+        """Returns an object of supported council's and their required fields."""
+        # Fetch the JSON data from the provided URL
+        url = "https://raw.githubusercontent.com/robbrad/UKBinCollectionData/master/uk_bin_collection/tests/input.json"
```
Comment (OliverCullimore) [inline custom_components/uk_bin_collection/config_flow.py:23]: We may need to bundle this file as part of the Python library build process to ensure we use a copy that matches the library's capabilities so the component doesn't try to offer Councils that aren't supported in the version that's currently installed?

**Review by OliverCullimore — CHANGES_REQUESTED** [review]

**Review by OliverCullimore — APPROVED** [review]

**Comment by robbrad** [comment]
Thanks @OliverCullimore !!

**Pull request merged**